tornadox/solve.py

```diff
@@ -59,6 +59,53 @@ def solve_ivp_for_terminal_value(*, f, df, tspan, u0, solver, **solver_kwargs):
     return extract_qoi_fn(t, state)
 
 
+@partial(jax.jit, static_argnames=("f", "df", "solver"))
+def solve_ivp_saveat(*, f, df, u0, solver, saveat, **solver_kwargs):
```

Comment:
if you have a better name than `solve_ivp_saveat`, please say so. I considered "checkpoint", but preferred `saveat` because it has less autodiff-connotation.

Reply:
No, I dont think so. Same discussion as for the attempt_step() vs attempt_step_forward_only(): The saveat-versions are strictly more expensive per step. (And it is also much more complicated in terms of implementation, because it nests all sorts of loops)

Reply:
Ah is it because a saveat implementation of the terminal value solver would then also attempt to smooth back all the way to the initial value? Didn't consider that, but I see that that would be more expensive.

Reply:
Not necessarily that (because the choice between smoothing, sampling, or doing something else is up to the user), but more so because it computes a bunch of smoothing gains that are not needed.